### PR TITLE
add lock_height to Transaction and TxKernel

### PR DESCRIPTION
**Pull request by antiochp**
Initial attempt at introducing "time locked" transactions.

------------------

TODO -
- [x] rename `transaction_fee_and_lock_height_to_bytes` -> `kernel_sig_msg`
- [x] add tests to cover adding invalid tx to pool
- [x] add tests to cover building invalid block

-----------------

* add `lock_height` to `Transaction` and `TxKernel`
* msg being signed in a kernel now includes both `fee` and `lock_height`
* verify on kernel checks both `fee` and `lock_height` against the signature
* wallet sender sets current chain height as the lock_height (as proof of concept)
* pool checks the tx is valid (in terms of lock_height) before adding to pool
* block is invalid if built with a tx with lock_height in future is included

Note: this is _similar_ to the `lock_height` on a coinbase output but not the same.
This is a "timelock" on the tx/kernel itself, not an output.

See #25 for more info/context/discussion.

### CODE REVIEW

**Comment by antiochp** [comment]
working on fixing up those `pool` test failures

File: core/src/core/transaction.rs [hunk 1]
```diff
@@ -453,19 +489,93 @@ impl ops::Add for SumCommit {
 	}
 }
 
-fn u64_to_32bytes(n: u64) -> [u8; 32] {
+/// TODO - consider putting this in consensus.rs so we can reuse it
+/// consistently across Transaction and TxKernel
+/// TODO - add some test coverage for this to make sure the bytes are in there
+/// correctly
+fn transaction_fee_and_lock_height_to_bytes(fee: u64, lock_height: u64) -> [u8; 32] {
```
Comment (ignopeverell) [inline core/src/core/transaction.rs]: I'd use a name more descriptive of what's produced rather than consumed. Maybe `kernel_sig_msg`?

**Comment by ignopeverell** [comment]
Good in principle. I'm a little bummed we have to have another clear value in addition to fees but I don't have a better option for now.

File: core/src/core/transaction.rs (hunk 1 above)
Comment (antiochp) [reply core/src/core/transaction.rs]: yeah good catch - that's on my todo list actually (named it that so I knew what was going on...)

**Comment by antiochp** [comment]
@ignopeverell any further feedback/thoughts on this?
I added some test coverage for both adding timelocked txs to the pool and building a block from timelocked txs.

**Comment by ignopeverell** [comment]
Looks good to me!

**Pull request merged**